homework3/task04/project_is_armstrong/task4_armstrong.py

```diff
@@ -0,0 +1,30 @@
+"""
+Task.
+
+Armstrong number is a number that is the sum of its own digits each raised to the power of the number of digits.
+https://en.wikipedia.org/wiki/Narcissistic_number
+
+Examples:
+
+- 9 is an Armstrong number, 9 = 9^1 = 9
+- 10 is not: 10 != 1^2 + 0^2 = 1
+- 153 is : 153 = 1^3 + 5^3 + 3^3 = 1 + 125 + 27 = 153
+
+
+Write a function that detects if a number is Armstrong number in functionaly style:
+ - use map or other utilities from functools library,
+ - use anonymous functions (or use function as argument)
+ - do not use loops, preferably using list comprehensions
+
+### Example function signature and call
+"""
+from functools import reduce
+
+
+def is_armstrong(number: int) -> bool:
+    """Return True if a number is Armstrong number."""
+    power = len(str(number))
+    result = reduce(
```

Comment:
Do you know why reduce was removed from the built-ins?

Reply:
reduce() makes code more complex for understanding. it can often lead to illegible code that's hard to maintain.

Reply:
And what we can do about it?

Reply:
Instead of reduce() we can use sum(), which simplify understanding what is going in our code. 
Like `return sum(int(char)**power for char in str(number)) == number`

Reply:
Let's use it.